# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 7b00a16ad21e
# Full Hash: 7b00a16ad21ee2b820b679199ae3806250d861e9
# Author: Nathan LaPre <nlapre@mozilla.com>
# Date: 2024-09-10 09:03:27
# Regressor Bug: 1794974
# File Overlap Count: 1
# Description:
#   Bug 1794974: Part 2: Add ability to query and send individual cache domains, r=Jamie
#   
#   This revision adds the capability of querying and sending information about
#   individual cache domains. It introduces the concept of active cache domains to
#   the accessibility service: cache domains that we think clients need. Conversely,
# ==============================================================================

diff -r 642b6d8c4d3b -r 7b00a16ad21e dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Mon Sep 09 23:02:19 2024 +0000
+++ b/dom/ipc/PContent.ipdl	Mon Sep 09 23:02:19 2024 +0000
@@ -764,15 +764,22 @@
     async UnlinkGhosts();
 
     /**
-     * Start accessibility engine in content process.
+     * Start accessibility engine in content process. aDomains specifies the
+     * cache domains for which content processes should send info.
      */
-    async ActivateA11y();
+    async ActivateA11y(uint64_t aDomains);
 
     /**
      * Shutdown accessibility engine in content process (if not in use).
      */
     async ShutdownA11y();
 
+    /**
+     * Set accessibility cache domains, effectively requesting more or less
+     * cached information from the content process.
+     */
+    async SetCacheDomains(uint64_t aCacheDomains);
+
     async AppInfo(nsCString version, nsCString buildID, nsCString name, nsCString UAName,
                   nsCString ID, nsCString vendor, nsCString sourceURL, nsCString updateURL);
 
